File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    dummy:=&ListNode{}
    cur:=dummy
    for list1 !=nil && list2 !=nil{
        if list1.Val>list2.Val{
            cur.Next=list2
            list2=list2.Next
        }else{
            cur.Next=list1
            list1=list1.Next
        }
        cur=cur.Next
    }
    if list1==nil{
        cur.Next=list2
    }
    if list2==nil{
        cur.Next=list1
    }
    return dummy.Next
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy=ListNode()
        cur=dummy # not copy, they point to the same inmemory adress
        while list1 or list2:
            if not list1:
                cur.next=list2
                break
            elif not list2:
                cur.next=list1
                break
            if list1.val>=list2.val:
                cur.next=ListNode(list2.val)
                list2=list2.next
            else:
                cur.next=ListNode(list1.val)
                list1=list1.next
            cur=cur.next
        return dummy.next